enroll/endpoint.go

```diff
@@ -5,19 +5,25 @@ import (
 	"golang.org/x/net/context"
 )
 
+type Endpoints struct {
+	GetEnrollEndpoint endpoint.Endpoint
+}
+
 type mdmEnrollRequest struct{}
 
 type mdmEnrollResponse struct {
 	Profile
 }
 
-func makeEnrollEndpoint(svc Service) endpoint.Endpoint {
+func MakeServerEndpoints(s Service) Endpoints {
+	return Endpoints{
+		GetEnrollEndpoint: MakeGetEnrollEndpoint(s),
+	}
+}
+
+func MakeGetEnrollEndpoint(s Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		//req := request.(mdmEnrollRequest)
-		profile, err := svc.Enroll()
-		if err != nil {
-			return mdmEnrollResponse{}, err
-		}
-		return mdmEnrollResponse{profile}, nil
+		profile, err := s.Enroll(ctx)
+		return mdmEnrollResponse{profile}, err
```

Comment:
The correct way to handle the error would be to add it to the response struct. This is because it's a business logic error(as opposed to a http transport error), and the server will encode the error response to the client. 
Add
```
Err     error `plist:"error,omitempty"
```
to the `mdmEnrollResponse` struct. 

I'm ok with merging this and fixing in a separate PR if you open an issue so we don't forget.